### PR TITLE
Agents

### PR DESCRIPTION
**Pull request by j0ni**
Totally open to not merging this if it seems overly indulgent - I won't be offended :)

Related: how best to manage this server's "progression" from a teaching perspective? Keep multiple branches to look at? Tag different versions on master?

### CODE REVIEW

**Comment by yogthos** [comment]
:+1:

File: src/chat_server/core.clj [hunk 1]
```diff
@@ -1,41 +1,95 @@
...
-        (do
-          (socket/write-line client "ERROR: Nick already taken")
-          (socket/close-socket client))))))
+  (let [{:keys [socket nick channels] :or {channels []}} @client]
+    (loop [line (socket/read-line socket)]
+      (let [[command & words] (string/split line #" ")]
+        (case command
+          "USER" (set-nick client (string/join "-" words))
+          "MSG" (send-message client (string/join " " words))
+          "QUIT" (terminate-client! client)
```
Comment (settinghead) [inline src/chat_server/core.clj:60]: Do we also need to free up the nickname registration here?

**Comment by j0ni** [comment]
That happens [here](https://github.com/ClojureTO/chat-server/blob/master/src/chat_server/core.clj#L47)

File: src/chat_server/core.clj (hunk 1 above)
Comment (j0ni) [inline src/chat_server/core.clj:60]: That happens [here](https://github.com/ClojureTO/chat-server/blob/master/src/chat_server/core.clj#L47)

Comment (settinghead) [inline src/chat_server/core.clj:60]: ah. get it now. thanks.

**Pull request merged**